File: Data Structures & Algorithms/maximum-subarray/submission-1.java
class Solution {
    public int maxSubArray(int[] nums) {
        int sum = 0;
        int max = nums[0];
        int i = 0;
        while (nums[i] < 0) {
            if (nums[i] > max) {
                max = nums[i];
            }
            i++;
            if (i == nums.length) {
                return max;
            }
        }
        for (; i < nums.length; i++) {
            sum += nums[i];
            if (sum < 0) {
                sum = 0;
            } else if (sum > max) {
                max = sum;
            }
        }
        return max;
    }
}

File: Data Structures & Algorithms/longest-common-subsequence/submission-1.py
class Solution:
    def longestCommonSubsequence(self, text1: str, text2: str) -> int:
        dp = [[0 for _ in range(0, len(text2) + 1)] for _ in range(0, len(text1) + 1)]
        for i in range(len(dp) - 2, -1, -1):
            for j in range(len(dp[0]) - 2, -1, -1):
                val = max(dp[i+1][j], dp[i][j+1])
                if text1[i] == text2[j]:
                    val = max(val, dp[i+1][j+1] + 1)
                dp[i][j] = val
        return dp[0][0]